cap/01-categorie.tex

```diff
@@ -1,9 +1,272 @@
+    $$
+    d_-(f_1(e)) = f_0(d_-(e)) ,\qquad d_+(f_1(e)) = f_0(d_+(e)) .
+    $$
+    Dimostra che $f_1:E\to E'$ associa a uno spigolo tra $x$ e $y$ uno spigolo tra $f_0(x)$ e $f_0(y)$. 
+    \item Dimostra che i multigrafi diretti e i loro omomorfismi formano una categoria. 
+    \item Una categoria piccola si può vedere come un multigrafo con identità e composizione: riscrivi la definizione di categoria (piccola) in termini di multigrafi.  In che modo questa costruzione generalizza la definizione di monoide (come insieme con identità e composizione)?
+    \item Un grafo diretto si può vedere come un multigrafo dove tra due vertici c'è al massimo uno spigolo. Dimostra che la costruzione al punto 3, se applicata ad un grafo diretto, dà un preordine. 
+\end{enumerate}
+
+
+\section{Isomorfismi}\label{isomorfismi}
+
+\begin{definition}
+ Dati due oggetti $X$ e $Y$ in una categoria $\ctC$, un \emph{isomorfismo} tra $X$ e $Y$ è una coppia di morfismi $f:X\to Y$ e $g:Y\to X$ tali che $g\circ f=\id_X$ e $f\circ g=\id_Y$. 
+ $f$ è chiamato l'\emph{inverso} di $g$, e a volte si indica con $g^{-1}$, e viceversa.
+ 
+ Se esiste un isomorfismo tra $X$ e $Y$ diciamo che i due oggetti sono \emph{isomorfi}.
+\end{definition}
+
+Può essere utile un'interpretazione grafica: intuitivamente, un isomorfismo è una situazione dove l'applicazione di $f$ si può invertire o annullare tramite $g$, e viceversa, e si può sempre passare da $X$ a $Y$ senza perdere informazioni. 
+$$
+\begin{tikzcd}
+ X \ar[shift left]{r}{f} & Y \ar[shift left]{l}{g}
+\end{tikzcd}
+$$
+
+Normalmente si indica la coppia $(f,g)$ solo con $f$ o con $g$, lasciando l'inverso implicito (l'inverso, se esiste, è unico, si vedano gli esercizi).
+
+\begin{examples}
+ La definizione di isomorfismo comprende molte delle nozioni di ``struttura equivalente'' usate in matematica:
+ \begin{itemize}
+  \item In $\ctSet$, gli isomorfismi sono le biiezioni, o corrispondenze biunivoche. In particolare, due insiemi sono isomorfi se e solo se hanno la stessa cardinalità.
+  \item In $\ctVect_\ictR$, gli isomorfismi sono le biiezioni lineari. In particulare, in dimensione finita, due spazi vettoriali sono isomorfi se e solo se hanno la stessa dimensione.
+  \item In $\ctTop$, gli isomorfismi sono gli omeomorfismi: funzioni continue con un'inversa continua. 
+  \item In $\ctGrp$, gli isomorfismi sono gli isomorfismi di gruppi. 
+ \end{itemize}
+\end{examples}
+
+\begin{warning}
+ Spesso una categoria viene chiamata con il nome dei suoi oggetti ($\ctVect$, $\ctGrp$, eccetera), ma è importante sapere \emph{quali sono i morfismi} per sapere che struttura in particolare si sta studiando. Questo è particolarmente importante nel caso degli isomorfismi: per esempio, $\ictR$ e $\ictR^2$, come insiemi, sono isomorfi, ma non come spazi vettoriali. Per cui, nella categoria $\ctVect$, gli oggetti $\ictR$ e $\ictR^2$ sono isomorfi, ma non sono isomorfi se li intendiamo come oggetti della categoria $\ctSet$. 
```

Comment:
L'esempio è al contrario: `$\ictR$` e `$\ictR^2$` sono isomorfi in `$\ctSet$` ma non in `$\ctVect$`.

Reply:
Un altro esempio che fa capire questo punto è la sottocategoria degli spazi topologici i cui oggetti sono R^n per i vari n naturali: la categoria ha esattamente gli stessi oggetti di Vect_R, ma i morfismi sono tutte le funzioni continue, quindi questa categoria si comporta molto come Top e niente affatto come Vect.

E' il caso di aggiungere che Ehresmann prendeva molto sul serio questa cosa che una categoria è definita dai suoi morfismi e chiamava "fonctions" la categoria Set, o Ehresmann non lo legge più nessuno oggi come oggi?

Reply:
Corretto.